neuralnetwork/genome.js

```diff
@@ -89,42 +89,54 @@ class Genome {
         };
     };
 
-    static copyNode = (node) => {
-        return { id: node.id, type: node.type, value: node.value };
-    };
-
-    static copyConnection = (connection) => {
-        return { in: Genome.copyNode(connection.in), 
-                 out: Genome.copyNode(connection.out), 
-                 weight: connection.weight, 
-                 isEnabled: connection.isEnabled, 
-                 innovation: connection.innovation };
-    };
-
     static crossover = (genomeA, genomeB) => {
 
+        let addedIds = [];
         let copiedNodes = [];
 
+        genomeA.nodeGenes.forEach(node => {
+            if (node !== undefined && !addedIds[node.id]) {
+                copiedNodes[node.id] = { ...node };
+                addedIds[node.id] = true;
+            }
+        });
 
+        genomeB.nodeGenes.forEach(node => {
+            if (node !== undefined && !addedIds[node.id]) {
+                copiedNodes[node.id] = { ...node };
+                addedIds[node.id] = true;
+            }
+        });
 
-        let innovationMap = {};
+        let innovationMap = [];
 
         genomeA.connectionGenes.forEach(connection => {
             if (innovationMap[connection.innovation] === undefined) {
                 innovationMap[connection.innovation] = [];
             }
-            innovationMap[connection.innovation].push(connection);
+            innovationMap[connection.innovation].push({ ...connection });
         });
 
         genomeB.connectionGenes.forEach(connection => {
             if (innovationMap[connection.innovation] === undefined) {
                 innovationMap[connection.innovation] = [];
             }
-            innovationMap[connection.innovation].push(connection);
+            innovationMap[connection.innovation].push({ ...connection });
         });
 
-        console.log(innovationMap)
+        let copiedConnections = [];
+
+        innovationMap.forEach(connectionList => {
```

Comment:
dont we want to choose the connections from the most fit genome? or are we starting assuming they are equal fitness